changedetectionio/update_worker.py

```diff
@@ -63,14 +66,13 @@ def run(self):
 
                                     # Update history with the stripped text for future reference, this will also mean we save the first
                                     # Should always be keyed by string(timestamp)
-                                    self.datastore.update_watch(uuid, {"history": {str(update_obj["last_checked"]): fname}})
-
-                                    watch = self.datastore.data['watching'][uuid]
+                                    timestamp = str(update_obj["last_checked"])
+                                    watch['history'].update({timestamp: fname})
 
                                     print (">> Change detected in UUID {} - {}".format(uuid, watch['url']))
 
                                     # Get the newest snapshot data to be possibily used in a notification
-                                    newest_key = self.datastore.get_newest_history_key(uuid)
+                                    newest_key = watch.newest_history_key
                                     if newest_key:
                                         with open(watch['history'][newest_key], 'r') as f:
                                             newest_version_file_contents = f.read().strip()
```

Comment:
can be
`watch.get_newest_snapshot_text()`

or 

`watch.get_snapshot_text(timestmap)`

Reply:
Removes a lot of repeated `.read()`